evennia/utils/tests/test_containers.py

```diff
@@ -11,64 +11,76 @@
+    @override_settings(GLOBAL_SCRIPTS={'script_name': {}})
+    def test_start_with_typeclassless_script(self):
+        """No specified typeclass should fallback to base"""
+        gsc = containers.GlobalScriptContainer()
 
-		self.assertEqual(len(gsc.typeclass_storage), 0)
+        gsc.start()
 
-	@override_settings(GLOBAL_SCRIPTS={'script_name': {}})
-	def test_start_with_typeclassless_script(self):
-		"""No specified typeclass should fallback to base"""
-		gsc = containers.GlobalScriptContainer()
+        self.assertEqual(len(gsc.typeclass_storage), 1)
+        self.assertIn('script_name', gsc.typeclass_storage)
+        self.assertEqual(gsc.typeclass_storage['script_name'], _BASE_TYPECLASS)
 
-		gsc.start()
+    @override_settings(GLOBAL_SCRIPTS={'script_name': {'typeclass': 'evennia.utils.tests.test_containers.NoScript'}})
+    def test_start_with_nonexistent_script(self):
+        """Missing script class should fall back to base"""
+        gsc = containers.GlobalScriptContainer()
 
-		self.assertEqual(len(gsc.typeclass_storage), 1)
-		self.assertIn('script_name', gsc.typeclass_storage)
-		self.assertEqual(gsc.typeclass_storage['script_name'], _BASE_TYPECLASS)
+        gsc.start()
 
-	@override_settings(GLOBAL_SCRIPTS={'script_name': {'typeclass': 'evennia.utils.tests.test_containers.NoScript'}})
-	def test_start_with_nonexistent_script(self):
-		"""Missing script class should fall back to base"""
-		gsc = containers.GlobalScriptContainer()
+        self.assertEqual(len(gsc.typeclass_storage), 1)
+        self.assertIn('script_name', gsc.typeclass_storage)
+        self.assertEqual(gsc.typeclass_storage['script_name'], _BASE_TYPECLASS)
 
-		gsc.start()
+    @override_settings(GLOBAL_SCRIPTS={'script_name': {'typeclass': 'evennia.utils.tests.test_containers.GoodScript'}})
+    def test_start_with_valid_script(self):
+        gsc = containers.GlobalScriptContainer()
 
-		self.assertEqual(len(gsc.typeclass_storage), 1)
-		self.assertIn('script_name', gsc.typeclass_storage)
-		self.assertEqual(gsc.typeclass_storage['script_name'], _BASE_TYPECLASS)
+        gsc.start()
 
-	@override_settings(GLOBAL_SCRIPTS={'script_name': {'typeclass': 'evennia.utils.tests.test_containers.GoodScript'}})
-	def test_start_with_valid_script(self):
-		gsc = containers.GlobalScriptContainer()
+        self.assertEqual(len(gsc.typeclass_storage), 1)
+        self.assertIn('script_name', gsc.typeclass_storage)
+        self.assertEqual(gsc.typeclass_storage['script_name'], GoodScript)
 
-		gsc.start()
+    @override_settings(GLOBAL_SCRIPTS={'script_name': {'typeclass': 'evennia.utils.tests.test_containers.InvalidScript'}})
+    def test_start_with_invalid_script(self):
+        """Script class doesn't implement required start methods"""
+        gsc = containers.GlobalScriptContainer()
 
-		self.assertEqual(len(gsc.typeclass_storage), 1)
-		self.assertIn('script_name', gsc.typeclass_storage)
-		self.assertEqual(gsc.typeclass_storage['script_name'], GoodScript)
```

Comment:
Maybe after `assertRaises`, it would be good to assert that `typeclass_storage` does not have `InvalidScript`. Likewise for the others that raise.

Reply:
I admit I'm fairly new to unit testing, but I don't understand what the purpose would be for checking the specific details of a failed state, particularly one that would halt execution of the entire process.

That is, if it raised the exception, there shouldn't *be* a `typeclass_storage` anymore to contain or not contain the script. It's a bit like checking which tabs your browser has open after it's crashed and isn't running.